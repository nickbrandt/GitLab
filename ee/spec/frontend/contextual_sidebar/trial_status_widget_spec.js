import { GlLink } from '@gitlab/ui';
import { shallowMount } from '@vue/test-utils';

import TrialStatusWidget from 'ee/contextual_sidebar/components/trial_status_widget.vue';

describe('TrialStatusWidget component', () => {
  let wrapper;

  const findGlLink = () => wrapper.findComponent(GlLink);

  const createComponent = (props = {}) => {
    return shallowMount(TrialStatusWidget, {
      propsData: {
        daysRemaining: 20,
        navIconImagePath: 'illustrations/golden_tanuki.svg',
        percentageComplete: 10,
        planName: 'Ultimate',
        plansHref: 'billing/path-for/group',
        ...props,
      },
    });
  };

  afterEach(() => {
    wrapper.destroy();
    wrapper = null;
  });

  describe('without the optional containerId prop', () => {
    beforeEach(() => {
      wrapper = createComponent();
    });

    it('matches the snapshot', () => {
      expect(wrapper.element).toMatchSnapshot();
    });

    it('renders without an id', () => {
      expect(findGlLink().attributes('id')).toBe(undefined);
    });

    it('renders with the correct tracking data attributes', () => {
      const attrs = findGlLink().attributes();
      expect(attrs['data-track-event']).toBe('click_link');
      expect(attrs['data-track-label']).toBe('trial_status_widget');
      expect(attrs['data-track-property']).toBe('experiment:show_trial_status_in_sidebar');
    });
  });

  describe('with the optional containerId prop', () => {
    beforeEach(() => {
      wrapper = createComponent({ containerId: 'some-id' });
    });

    it('renders with the given id', () => {
      expect(findGlLink().attributes('id')).toBe('some-id');
    });
  });
});
