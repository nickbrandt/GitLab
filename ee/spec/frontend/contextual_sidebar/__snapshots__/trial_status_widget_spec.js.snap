// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrialStatusWidget component without the optional containerId prop matches the snapshot 1`] = `
<gl-link-stub
  data-track-event="click_link"
  data-track-label="trial_status_widget"
  data-track-property="experiment:show_trial_status_in_sidebar"
  href="billing/path-for/group"
  title="Ultimate Trial – 20 days left"
>
  <div
    class="gl-display-flex gl-flex-direction-column gl-align-items-stretch gl-w-full"
  >
    <span
      class="gl-display-flex gl-align-items-center"
    >
      <span
        class="nav-icon-container svg-container"
      >
        <img
          class="svg"
          src="illustrations/golden_tanuki.svg"
          width="16"
        />
      </span>
       
      <span
        class="nav-item-name gl-white-space-normal"
      >
        
        Ultimate Trial – 20 days left
      
      </span>
    </span>
     
    <span
      class="gl-display-flex gl-align-items-stretch gl-mt-3"
    >
      <gl-progress-bar-stub
        class="gl-flex-grow-1"
        value="10"
      />
    </span>
  </div>
</gl-link-stub>
`;
